sentinelsat/sentinel.py

```diff
@@ -145,13 +144,13 @@ def query(self, area, initial_date=None, end_date=datetime.now(), **keywords):
         and any other search keywords accepted by the SciHub API.
         """
         query = self.format_query(area, initial_date, end_date, **keywords)
-        self.load_query(query)
-        return self.products
+        return self.load_query(query)
 
-    def load_query(self, query, start_row=0):
+    def load_query(self, query, start_row=0, products=[]):
         """Do a full-text query on the SciHub API using the format specified in
            https://scihub.copernicus.eu/twiki/do/view/SciHubUserGuide/3FullTextSearch
         """
+        output = products
```

Comment:
```python
def load_query(self, ..., products=[]):
    output = products
    ...
    output += other_products
```
This will cause bugs. Python's default parameter values are shared across function calls, which is bad news for mutable objects like lists: http://docs.python-guide.org/en/latest/writing/gotchas/.

Consider doing
```python
output = []
output += products
```
instead to avoid this issue.